stumpy/core.py

```diff
@@ -2561,6 +2575,201 @@ def _select_P_ABBA_value(P_ABBA, k, custom_func=None):
     return MPdist
 
 
+@njit
+def _merge_topk_PI(PA, PB, IA, IB):
+    """
+    Merge two top-k matrix profiles `PA` and `PB`, and update `PA` (in place).
+    When the inputs are 1D arrays, PA[i] is updated if it is less than PB[i] and
```

Comment:
Oops! You are right! Thanks for catching it! Will be fixed.

Reply:
I should have used different names for indices in description of 1D case. Will be fixed. Thanks!!

Reply:
@seanlaw 
Just let you know, I think I was wrong about changing the index. I think `PA[i], PB[i]` is correct. Because we are trying to update the (first) NN of the i-th subseq. So, we need  to see if its NN is `IA[i]` or `IB[i]`. And, their corresponding distances (i.e. distance between `i` and `IA[i]` / `IB[i]`) are stored in `PA[i]` and PB[i] respectively.

Reply:
Ahh, gotcha. I missed the "When the inputs are 1D arrays" part. This makes sense now. My brain was still thinking about the 2D case